sw/device/silicon_creator/rom_exts/docs/manifest.md

```diff
@@ -8,457 +8,87 @@ aliases:
-signing and validation systems are implementing, so that alternate
-implementations may be produced if necessary.
-
-Notation:
-
-*   `rom_ext` - ROM_EXT Image, including manifest and contents (Variable Length)
-    `signed_area(rom_ext)` denotes just the area between the start of the signed
-    contents and the end of the image.
-*   `device_usage_value` - Device-Calculated Usage Constraint Value (1024
-    bytes). This is the "device usage value" described above, as calculated by
-    the Mask ROM.
-*   `system_state_value` - Device System State Value (32 bytes). This is a Mask
-    ROM calculated value that summarises the current device state.
-*   `private_key` - RSA Private Key
-*   `public_key` - RSA Public Key
-*   `signature` - ROM_EXT RSA Signature,
-*   `||` - Denotes concatenation without a delimiter.
-*   `:=` - Denotes assignment.
-
-## Algorithms
+*   `signature`: [`RSASSA-PKCS1-v1_5`][rsassa_pkcs1_v1_5] signature of the
+    image generated using a 3072-bit RSA private key and the SHA-256 hash
+    function. The signed region of an image starts immediately after this
+    field and extends to the end of the image.
 
-*   We primarily use [`RSASSA-PKCS1-V1_5-SIGN`][RSASSA-PKCS1-V1_5-SIGN] and
-    [`RSASSA-PKCS1-V1_5-VERIFY`][RSASSA-PKCS1-V1_5-VERIFY] for signatures and
-    verification.
-*   RSA Signing and Verification is done with 3072-bit keys.
-*   The `Hash` operation may be done with SHA2-256, SHA3-256, SHA3-384, or
-    SHA3-512, depending on the Signature Algorithm Identifier provided in the
-    manifest. The `Hash` operation is required to provide a security level
-    equivalent to that provided by the signature.
+*   `modulus`:  Modulus of the signer's 3072-bit RSA public key.
 
-## Signing
+*   `exponent`: Exponent of the signer's RSA public key. The only values
+    supported by OpenTitan are 3 and 65537.
 
-This is usually done on a host machine, which does not have access to a real OT
-device. Thus there has to be a way for the host signer to predict the value of
-`device_usage_value` and `system_state_value`, as they would be calculated
-on-device.
+*   `identifier`: Image identifier used to identify boot stage images. The
+    value of this field must be `0x4552544f` (ASCII: "OTRE") for a `ROM_EXT`
+    image and `0x3042544f` (ASCII: "OTB0") for the first owner stage.
 
-The `private_key` is never available on-device, in this scheme.
+*   `length`: Length of the image (including the manifest) in bytes.
 
-```
-system_state_value := predict_system_state_value()
-device_usage_value := predict_device_usage_value(rom_ext.usage_constraint)
+*   `version_major`: Major version of the image.
 
-message := system_state_value || device_usage_value || signed_area(rom_ext)
-signature := RSASSA-PKCS1-V1_5-SIGN(private_key, message)
-```
+*   `version_minor`: Minor version of the image.
 
-## Validation
+*   `security_version`: Security version of the image used for anti-rollback
```

Comment:
Do we have any guidance as to what to set this to?

Reply:
This is a monotonically increasing integer.  It could be a timestamp, a CL number (if your VCS supports such things) or a manually managed integer.

It exists to implement anti-rollback policy and divorces the anti-rollback policy from the formal version number.  All customer visible version numbering schemes eventually get captured by marketing folks who will apply their own numerology, thus ruining any scheme you thought you might implement with your version numbers.

Reply:
Added a sentence that says this must be a monotonically increasing integer.